### PR TITLE
Clean up batch process block

### PR DESCRIPTION
**Pull request by terencechain**
This PR did the following:
- Clean up `savePostStateInfo`. It doesn't require the boolean flag anymore, and it's never used for initial syncing. 
- Move `saveInitSyncBlock` to after `notifyNewPayload` passes. We shouldn't cache the initial sync block without knowing the outcome of payload validation 
- Call `notifyForkchoiceUpdate` once per batch, the last of the batch block. I don't see why BN needs to notify EE on every block for the forkchoice update. One per block batch is more than enough

### CODE REVIEW

**Review by prestonvanloon — APPROVED** [review]

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -601,12 +601,10 @@ func getBlockPayloadHash(blk block.BeaconBlock) ([32]byte, error) {
 
 // This saves post state info to DB or cache. This also saves post state info to fork choice store.
 // Post state info consists of processed block and state. Do not call this method unless the block and state are verified.
-func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b block.SignedBeaconBlock, st state.BeaconState, initSync bool) error {
+func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b block.SignedBeaconBlock, st state.BeaconState) error {
```
Comment (potuz) [inline beacon-chain/blockchain/process_block.go:604]: As of now this function  only saves the block and states and for this we are copying the state and moving it around.  Does it make sense to just do these two things in `onBlock` and remove this function?

**Review by potuz — COMMENTED** [review]
Should we have a test that the init sync block is not being saved on `onBlock`? Or that we only call fcu on the last block of the batch? The first one seems very simple

File: beacon-chain/blockchain/process_block.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/blockchain/process_block.go:604]: Probably but let's do that when we clean up `onBlock` 🙏🏼

**Comment by terencechain** [comment]
> Should we have a test that the init sync block is not being saved on `onBlock`? Or that we only call fcu on the last block of the batch? The first one seems very simple

inial sync block is never saved `onBlock` anyway. It's passed in `false`

**Pull request merged**